src/reducers/existingThoughtChange.js

```diff
@@ -68,13 +69,34 @@ export const existingThoughtChange = (state, { oldValue, newValue, context, show
       .removing if the old path is already in the object
       .Checking for all the descendants and updating their path too
       .limiting number of recenlty edited thoughts
-      .Adding new thought to the array */
+      .Adding new thought to the array
+      .merging the paths with common majority subcontext
+      .concating recently edited thought path only if no merge has happened
+
+  */
+
+  let isMerged = false // eslint-disable-line fp/no-let
```

Comment:
Mutation *may* be justifiable here, but let's see how the algorithm shakes out after the other requested changes are made.

Reply:
Sure